oshmem/mca/spml/ucx/spml_ucx.c

```diff
@@ -771,6 +772,16 @@ int mca_spml_ucx_get_nb(shmem_ctx_t ctx, void *src_addr, size_t size, void *dst_
     status = ucp_get_nbi(ucx_ctx->ucp_peers[src].ucp_conn, dst_addr, size,
                      (uint64_t)rva, ucx_mkey->rkey);
 
+    if (OPAL_UNLIKELY(mca_spml_ucx.nb_get_progress_thresh &&
```

Comment:
maybe it is worth to define a new  get and put functions with this code and update func pointers in `mca_spml_ucx_t` if `nb_(get|put)_progress_thresh` is not zero?
Then there will be no extra branch if this feature is not enabled.

Reply:
good idea, I'll change it, avoids an extra conditional

Reply:
Juist fyi, I updated the code, split the functionality to a new nb put/get call with progress and update the function pointer in case progress is used.

Reply:
Note that as far as I know calling function by pointer is similar to a branch from the CPU perspective. Might be a bit faster though as it is more stable.
So not much gain probably

Reply:
But as I’ve told @janjust I’m fine either way

Reply:
it is already called by function pointer, but we can use another function if periodical progress is needed

Reply:
Got it!